### DISCUSSION TITLE
Reverse Proxy infront of T-Pot

### DISCUSSION THREAD

**Discussion by Rapter1099** [telekom-security/tpotce#1249] [Q&A] [answered]
🗨️ Please post your questions in [Discussions](https://github.com/telekom-security/tpotce/discussions) and keep the issues for **issues**. Thank you 😁.<br>

Before you post your issue make sure it has not been answered yet and provide `basic support information` if you come to the conclusion it is a new issue.

- 🔍 Use the [search function](https://github.com/dtag-dev-sec/tpotce/issues?utf8=%E2%9C%93&q=) first
- 🧐 Check our [WIKI](https://github.com/dtag-dev-sec/tpotce/wiki)
- 📚 Consult the documentation of 💻 [Debian](https://www.debian.org/doc/), 🐳 [Docker](https://docs.docker.com/), the 🦌 [ELK stack](https://www.elastic.co/guide/index.html) and the 🍯 [T-Pot Readme](https://github.com/dtag-dev-sec/tpotce/blob/master/README.md).
- **⚠️ Provide [basic support information](#info) or similiar information with regard to your issue or we can not help you and will close the issue without further notice**

<br>
<br>
<br>

<a name="info"></a>
## ⚠️ Basic support information (commands are expected to run as `root`)

- What version of the OS are you currently using `lsb_release -a` and `uname -a`?
Distributor ID: Debian
Description:    Debian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye

Linux enchantingauthorization 5.10.0-20-amd64 #1 SMP Debian 5.10.158-2 (2022-12-13) x86_64 GNU/Linux

- What T-Pot version are you currently using? T-Pot 22.04.0
- What edition (Standard, Nextgen, etc.) of T-Pot are you running? Standard
- What architecture are you running on (i.e. hardware, cloud, VM, etc.)? Hardware
- Did you have any problems during the install? If yes, please attach `/install.log` `/install.err`. No
- How long has your installation been running? < 1 day
- Did you install upgrades, packages or use the update script? No
- Did you modify any scripts or configs? If yes, please attach the changes. No
- Please provide a screenshot of `glances` and `htop`.  N/A
- How much free disk space is available (`df -h`)? N/A
- What is the current container status (`dps.sh`)?
- What is the status of the T-Pot service (`systemctl status tpot`)?  N/A
- What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `netstat -tulpen`  N/A
- If a single container shows as `DOWN` you can run `docker logs <container-name>` for the latest log entries  N/A

I have been trying to put a reverse proxy in front of my TPot IP (to be accessed internally only, its not accessible via the internet, I just want to be able to use a URL instead of a IP:Port). I have managed to do this successfuly with cockpit, forwarding to IP:64294.
However when I have the same config and I forward to IP:64297 I get a 502 error.

Is anyone aware of a reason for this? Is it to do with the web auth that is on the webui? If so, how can I get around it?

Cheers!

**Comment by t3chn0m4g3** [comment, marked as answer]
Since some of the T-Pot services are [reverse-proxied](https://github.com/telekom-security/tpotce#services) already I am guessing you need to take that into account for your setup.

**Comment by cpyix** [comment]
I have no problems using haproxy. I had similar issues using nginx beforehand.
